src/app/modules/convai-checker/perspectiveapi.service.ts

```diff
@@ -130,11 +129,26 @@ export class PerspectiveApiService {
         console.error('No server url specified for a non-direct API call.'
                       + ' Defaulting to current hosted address');
       }
-      const headers = new HttpHeaders();
-      headers.append('Content-Type', 'application/json');
 
-      return this.httpClient.post(
-        serverUrl + '/suggest_score', data, {headers});
+      return this.sendRequest(
+          data, `${serverUrl}/suggest_score`, 'suggestScore');
     }
   }
+
+  private sendRequest<T, R>(data: T, endpoint: string, action: string):
+    Observable<R> {
+    return this.recaptchaV3Service === undefined ||
+      this.recaptchaV3Service === null ?
```

Comment:
Optional: I also love the ternary expression, but since the second option spans 3 lines, it might be easier to read as an if/else.

Reply:
Done.